Pkg 4115 v12.4 initial

danpetry opened the pull request:
**Destination channel:** defaults

### Links

- [PKG-4584](https://anaconda.atlassian.net/browse/PKG-4584)

### Explanation of changes:

- Fifth package of CUDA 12.4 build, following [conda-forge's pattern](https://github.com/conda-forge/staged-recipes/issues/21382) which was developed in conjunction with Nvidia. I don't see a need to stage these then release them all at once. They won't cause a problem released one at a time.
- Please check the main branch too, this has just been forked
- Changes made to adjust feedstocks for differences between conda-forge and defaults. Please see commit messages for description of changes.
- There's an [open issue](https://github.com/conda-forge/cuda-nvcc-feedstock/issues/40) upstream but I suggest we don't wait on it and catch it with the next update.


[PKG-4584]: https://anaconda.atlassian.net/browse/PKG-4584?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

danpetry committed:
Unpin the gcc version

It looks like our aggregate-pinned version, 11, is fine too:
https://docs.nvidia.com/cuda/archive/12.3.0/cuda-installation-guide-linux/index.html#host-compiler-support-policy

danpetry committed:
Adjust supported glibc version on linux-aarch64; it's different from conda-forge's
m

danpetry committed:
Skip s390x, CUDA doesn't support it

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_43i33p70s5/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:154: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:154: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:154: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

danpetry committed:
Don't cross-compile, even on linux-64

danpetry committed:
Symlink ar so CMake and nvcc find it

linux-aarch64 isn't built on on conda-forge, and the ar symlink is added to make
the tests pass on that platform.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_add0uvk5ik/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

SC426 commented:
You have 3 different `about` sections. Is this necessary? You also need a `license_family` and `dev_url`

danpetry commented:
These cuda feedstocks are deliberately kept as close as possible to conda-forge's, so that it's as easy as possible to update them when conda-forge bump their version. To reduce the large amount of work that such updates might generate

JeanChristopheMorinPerso commented:
@SC426 the different about sections are for each output. You can define an about section on a per output basis. That's totally valid. As for `dev_url`, there is no such thing because cuda toolkit is a closed sourced (proprietary) set of libraries.

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_844rvdrrzw/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_51eo6094yh/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_34pq2kmtet/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_5fvhyi1a7o/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
===== Final Report: =====
2 Errors and 1 Warning were found

</details>

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
ERROR conda.cli.main_run:execute(124): `conda run conda-lint /tmp/abs_a0t28ue1np/clone` failed. (See above for error)
The following problems have been found:

===== WARNINGS =====
- clone/recipe/meta.yaml:157: license_file_overspecified: Using license_file and license_url is overspecified.

===== ERRORS =====
- clone/recipe/meta.yaml:157: missing_dev_url: The recipe is missing a dev_url
- clone/recipe/meta.yaml:157: missing_license_family: The recipe is missing the `about/license_family` key.
===== Final Report: =====
2 Errors and 1 Warning were found

</details>
